### PR TITLE
Implement remaining random methods through ATen

### PR DESCRIPTION
**Pull request by colesbury**
I've renamed `means` to `mean` to be consistent with the variants which take `double`. Eventually, we should combine these signatures.

The in-place `bernoulli_` function is implemented via `copy_`.

### CODE REVIEW

File: tools/autograd/derivatives.yaml [hunk 1]
```diff
@@ -422,6 +426,16 @@
...
 
+- name: normal(Tensor mean, double std, Generator generator)
+  mean: grad.type().zeros(mean.sizes())
+
+- name: normal(double mean, Tensor std, Generator generator)
+  std: grad.type().zeros(std.sizes())
+
+- name: normal(Tensor mean, Tensor std, Generator generator)
+  mean: grad.type().zeros(mean.sizes())
+  std: grad.type().zeros(std.sizes())
```
Comment (apaszke) [inline tools/autograd/derivatives.yaml]: I'd say that it's surprising that these functions are differentiable. They either shouldn't be, or should be implemented as native reparametrized functions.

Comment (colesbury) [reply tools/autograd/derivatives.yaml]: This is equivalent to the previous code in `variable.py`

Comment (colesbury) [reply tools/autograd/derivatives.yaml]: (Setting the result to not require grad is equivalent to pushing back zeros)

Comment (apaszke) [reply tools/autograd/derivatives.yaml]: It's not really equivalent. Some operations are illegal on Variables that require grad, and they would work previously

**Review by gchanan — APPROVED** [review]
lgtm.  I think it's fine to abstain on the zeros_like vs not requiring grad issue for the purpose of this PR because the other random functions that dispatch through ATen behave this way already.

File: tools/autograd/derivatives.yaml (hunk 1 above)
Comment (colesbury) [reply tools/autograd/derivatives.yaml]: Btw, I like the idea re-parameterizing things like normal() (i.e. the VAE trick).  It seems like a better default than zero or raising exceptions

Comment (apaszke) [reply tools/autograd/derivatives.yaml]: The problem is you can't do this for all distributions, so we should probably never backprop through these (and set the results to requires_grad=False, to make it clear that they don't give meaningful gradients). If someone wants reparametrized gradients, use `torch.distributions`
